plugins/Jobs.cpp

```diff
@@ -160,33 +160,72 @@ bool BedrockPlugin_Jobs::peekCommand(SQLite& db, BedrockCommand& command) {
+        list<STable> jsonJobs;
+        if (SIEquals(request.methodLine, "CreateJob")) {
+            verifyAttributeSize(request, "name", 1, MAX_SIZE_SMALL);
+            jsonJobs.push_back(request.nameValueMap);
+        } else {
+            list<string> multipleJobs;
+            multipleJobs = SParseJSONArray(request["jobs"]);
+            for (auto& job : multipleJobs) {
+                if (!SContains(job, "name")) {
+                    throw "402 Missing name";
+                }
+                jsonJobs.push_back(SParseJSONObject(job));
+            }
         }
 
-        // Verify unique
-        SQResult result;
-        SINFO("Unique flag was passed, checking existing job with name " << request["name"]);
-        if (!db.read("SELECT jobID, data "
-                     "FROM jobs "
-                     "WHERE name=" + SQ(request["name"]) + ";",
-                     result)) {
-            throw "502 Select failed";
-        }
-        // If there's no job, or the existing job doesn't match the data we've been passed, escalate to master.
-        if (result.empty() || result[0][1] != request["data"]) {
-            return false;
-        }
+        for (auto& job : jsonJobs) {
+            // If no priority set, set it
+            int64_t priority = SContains(job, "priority") ? SToInt(job["priority"]) : JOBS_DEFAULT_PRIORITY;
 
-        // Supposed to be unique but not; notify the caller and return that we
-        // are processed
-        SINFO("Job already existed and unique flag was passed, reusing existing job " << result[0][0]);
-        content["jobID"] = result[0][0];
-        return true;
+            // We'd initially intended for any value to be allowable here, but for
+            // performance reasons, we currently will only allow specific values to
+            // try and keep queries fast. If you pass an invalid value, we'll throw
+            // here so that the caller can know that he did something wrong rather
+            // than having his job sit unprocessed in the queue forever. Hopefully
+            // we can remove this restriction in the future.
+            if (priority != 0 && priority != 500 && priority != 1000) {
+                throw "402 Invalid priority value";
+            }
+
+            // Validate that the parentJobID exists and is in the right state if one was passed.
+            int64_t parentJobID = SContains(job, "parentJobID") ? SToInt(job["parentJobID"]) : 0;
+            if (parentJobID) {
+                SQResult result;
+                if (!db.read("SELECT state FROM jobs WHERE jobID=" + SQ(parentJobID) + ";", result)) {
+                    throw "502 Select failed";
+                }
+                if (result.empty()) {
+                    throw "404 parentJobID does not exist";
+                } else if (!SIEquals(result[0][0], "RUNNING") && !SIEquals(result[0][0], "PAUSED")) {
```

Comment:
I think we should make this an if instead of else/if for consistency with the code around it.
```
if (C) {
  throw;
}

if (C2) {
  throw;
}

if (C3) {
  throw;
}

// as opposed to
if (C) {
  throw;
}

if (C2) {
  throw;
} else if (C3) {
  throw;
}
```